src/controllers/transactionController.ts

```
@@ -28,15 +28,11 @@ export async function listMyTransactions(
     if (!userId) {
-      res.status(401).json({ error: "User-scoped API key required" });
-      return;
+      throw new AppError("User-scoped API key required", 401, "UNAUTHORIZED");
     }
 
-    const query = listTransactionsQuerySchema.safeParse(req.query);
     if (!query.success) {
-      const msg = query.error.errors.map((x) => x.message).join("; ");
-      res.status(400).json({ error: msg });
-      return;
+      throw new AppError("Invalid query parameters", 400, "VALIDATION_ERROR", query.error.flatten());
```

ESLint

[error] Replace `"Invalid·query·parameters",·400,·"VALIDATION_ERROR",·query.error.flatten()` with `⏎········"Invalid·query·parameters",⏎········400,⏎········"VALIDATION_ERROR",⏎········query.error.flatten(),⏎······`

(prettier/prettier)